sjsonnet/src/sjsonnet/Std.scala

```diff
@@ -474,26 +472,24 @@ class Std(private val additionalNativeFunctions: Map[String, Val.Builtin] = Map.
       Val.Str(pos, str.asString.replaceAll(from.asString, to.asString))
     override def specialize(args: Array[Expr]) = args match {
       case Array(str, from: Val.Str, to) =>
-        try { (new SpecFrom(Pattern.compile(from.value)), Array(str, to)) } catch { case _: Exception => null }
+        try { (new SpecFrom(from.value), Array(str, to)) } catch { case _: Exception => null }
       case _ => null
     }
-    private class SpecFrom(from: Pattern) extends Val.Builtin2("strReplaceAll", "str", "to") {
+    private class SpecFrom(from: String) extends Val.Builtin2("strReplaceAll", "str", "to") {
       def evalRhs(str: Val, to: Val, ev: EvalScope, pos: Position): Val =
-        Val.Str(pos, from.matcher(str.asString).replaceAll(to.asString))
+        Val.Str(pos, Platform.getPatternFromCache(from).matcher(str.asString).replaceAll(to.asString))
```

Comment:
This is still defeating some of the point of the specialization: in these specialized ones we should be doing the pattern retrieval at the `def specialize` call and be using it in the body.

Reply:
Not sure I fully understand what the benefit of building the pattern in the constructor is.
in all cases, I'm trying to avoid explicitly importing Pattern in Std (due to the different implementations - which sadly don't even share an explicit common interface)

What if I do:
```
    private class SpecFrom(from: String) extends Val.Builtin2("strReplaceAll", "str", "to") {
      private val fromPattern = Platform.getPatternFromCache(from)
      def evalRhs(str: Val, to: Val, ev: EvalScope, pos: Position): Val =
        Val.Str(pos, fromPattern.matcher(str.asString).replaceAll(to.asString))
    }
```

Reply:
Ah, I think that works. You could also make it `private[this]` so it's just a regular private field, too.